source/Ship.cpp

```diff
@@ -2645,12 +2739,158 @@ bool Ship::IsParalyzed() const
 
 bool Ship::IsDisabled() const
 {
-	if(!isDisabled)
+	return isDisabled;
+}
+
+
+
+bool Ship::CalculateIsDisabled() const
+{
+	return hull < minimumHull || (!crew && RequiredCrew()) || IsOutOfEnergy();
+}
+
+
+
+// Determine if the ship has any usable weapons.
+bool Ship::IsArmed() const
+{
+	return IsArmed(false);
+}
+
+
+
+// Determine if the ship has any usable weapons against enemies.
+bool Ship::IsArmed(bool includingAntiMissile) const
+{
+	return (includingAntiMissile) ? (maxWeaponRange || hasAntiMissile) : maxWeaponRange;
+}
+
+
+
+// Check for enemies in the in the current system if ship is owned by player.
+bool Ship::IsEnemyInEscortSystem() const
+{
+	if(!IsYours())
 		return false;
+	return isEnemyInEscortSystem;
+}
+
+
 
-	double minimumHull = MinimumHull();
-	bool needsCrew = RequiredCrew() != 0;
-	return (hull < minimumHull || (!crew && needsCrew));
+// If a ship is nearly out of battery energy and is slow to charge or no charge,
+// then consider it low on energy.
+bool Ship::IsEnergyLow() const
+{
+	double frames = 60.;
+	double idleEnergy = frames * GetIdleEnergyPerFrame();
+	double maxEnergy = Attributes().Get("energy capacity");
+	double totalConsumption = frames * GetEnergyConsumptionPerFrame();
+	double currentEnergy = (CanBeCarried() && GetSystem() && !landingPlanet) ? GetCurrentEnergy() : maxEnergy * .75;
+	//double currentEnergy = GetCurrentEnergy();
```

Comment:
why was this commented this way?

and why do we sometimes take CurrentEnergy and sometimes max energy * .75 (which, shouldnt we take current energy * 0.75?)